rollup.config.mjs

```diff
@@ -13,7 +13,7 @@ const tsConfig = {
   module: 'esnext',
   moduleResolution: 'node',
   removeComments: true,
-  lib: ['es2021'],
+  lib: ['es2021', 'ES2022.Error'],
```

Comment:
This seems unrelated to the ticket, just curious as to the reasoning for this change.

Reply:
As a part of this work we were able to move UTF validation below the latin optimization (because below below 127 cannot be in an invalid format). On the webByteUtils side, validation is done by TextDecoder, but we have an API expectation that we throw a BSONError, but I figured it would be nice to maintain the original error from the TextDecoder. So I added support for the cause property on BSONError, TS doesn't allow us to pass a second argument to the Error constructor without adding this argument to the lib setting. 

This doesn't have downstream effects, we only support versions of Node.js that support the cause argument, but if this was run on old versions, it would just be ignored.